src/Razor/src/Microsoft.AspNetCore.Razor.Language/DefaultRazorIntermediateNodeLoweringPhase.cs

```diff
@@ -1728,14 +1728,21 @@ public override void VisitMarkupTagHelperAttribute(MarkupTagHelperAttributeSynta
                             return TagHelperMatchingConventions.CanSatisfyBoundAttribute(attributeName, a);
                         });
 
+                        var associatedAttributeParameterDescriptor = associatedAttributeDescriptor.BoundAttributeParameters.FirstOrDefault(p =>
+                        {
+                            return TagHelperMatchingConventions.SatisfiesBoundAttributeWithParameter(attributeName, associatedAttributeDescriptor, p);
+                        });
+
                         var setTagHelperProperty = new TagHelperPropertyIntermediateNode()
                         {
                             AttributeName = attributeName,
                             BoundAttribute = associatedAttributeDescriptor,
+                            BoundAttributeParameter = associatedAttributeParameterDescriptor,
                             TagHelper = associatedDescriptor,
                             AttributeStructure = node.TagHelperAttributeInfo.AttributeStructure,
                             Source = BuildSourceSpanFromNode(attributeValueNode),
                             IsIndexerNameMatch = TagHelperMatchingConventions.SatisfiesBoundAttributeIndexer(attributeName, associatedAttributeDescriptor),
+                            IsParameterMatch = associatedAttributeParameterDescriptor != null,
```

Comment:
Might as well just make this part of the `TagHelperPropertyIntermediateNode.IsParameterMatch`  getter and remove the setter